mm2src/gui_storage/src/account/storage/sqlite_storage.rs

```diff
@@ -0,0 +1,580 @@
+use crate::account::storage::{AccountStorage, AccountStorageError, AccountStorageResult};
+use crate::account::{AccountId, AccountInfo, AccountType, AccountWithCoins, AccountWithEnabledFlag, EnabledAccountId,
+                     EnabledAccountType, HwPubkey, MAX_ACCOUNT_DESCRIPTION_LENGTH, MAX_ACCOUNT_NAME_LENGTH,
+                     MAX_TICKER_LENGTH};
+use async_trait::async_trait;
+use db_common::sql_build::*;
+use db_common::sqlite::rusqlite::types::Type;
+use db_common::sqlite::rusqlite::{Connection, Error as SqlError, Result as SqlResult, Row};
+use db_common::sqlite::SqliteConnShared;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use mm2_number::BigDecimal;
+use std::collections::BTreeMap;
+use std::str::FromStr;
+use std::sync::{Arc, MutexGuard};
+
+const DEVICE_PUBKEY_MAX_LENGTH: usize = 20;
+const BALANCE_MAX_LENGTH: usize = 255;
+
+mod account_table {
```

Comment:
It looks like `account_coins_table` and `enabled_account_table` are child tables for `account_table`. Maybe adding a foreign key to `account_coins_table` and `enabled_account_table` with cascading on delete, etc.., can help avoiding multiple checks for if the account exists on the parent key among other things. What do you think?

Reply:
I like this idea
> It looks like account_coins_table and enabled_account_table are child tables for account_table. Maybe adding a foreign key to account_coins_table and enabled_account_table with cascading on delete, etc..

But it will not help to avoid the checks if the account exists or not on [AccountStorage::activate_coins](https://github.com/KomodoPlatform/atomicDEX-API/blob/gui-storage/mm2src/gui_storage/src/account/storage/sqlite_storage.rs#L503) because as I said here https://github.com/KomodoPlatform/atomicDEX-API/pull/1331#discussion_r961592855, we need to determine if the account exists or not. I don't like an idea to parse the SQL error string.
>  can help avoiding multiple checks for if the account exists on the parent table among other things.

Anyway, I think adding primary and foreign keys is a good idea because we can delete all coins from `account_coins_table` by deleting one account from `account_table`.

Reply:
> But it will not help to avoid the checks if the account exists or not on [AccountStorage::activate_coins](https://github.com/KomodoPlatform/atomicDEX-API/blob/gui-storage/mm2src/gui_storage/src/account/storage/sqlite_storage.rs#L503) because as I said here https://github.com/KomodoPlatform/atomicDEX-API/pull/1331#discussion_r961592855, we need to determine if the account exists or not. I don't like an idea to parse the SQL error string.

It's ok to keep the checks for if an account exists I guess, my only concern with this was that it's an unnecessary database call. But if you think it's essential, then it's fine :)